contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -6618,3 +6688,1155 @@ get_domain_typmodin(Type typ)
 	}
 	return typmodin;
 }
+
+/*
+ * Checks if a view is a dummy view created during object drop processing
+ *
+ * A dummy view has a specific structure:
+ * - SELECT command type
+ * - Empty FROM clause (no tables referenced)
+ * - No WHERE clause
+ * - Target list contains only constants
+ *
+ * Parameters:
+ * - viewOid: OID of the view to check
+ *
+ * Returns:
+ * - true if the view is a dummy view, false otherwise
+ */
+static bool
+is_dummy_view(Oid viewOid)
+{
+	Relation viewRel;
+	Query *viewQuery = NULL;
+	bool is_dummy = false;
+
+	viewRel = relation_open(viewOid, AccessShareLock);
+	
+	if (viewRel->rd_rules && viewRel->rd_rules->rules)
+	{
+		RewriteRule *rule = viewRel->rd_rules->rules[0];
+		
+		if (rule->actions && list_length(rule->actions) == 1)
```

Comment:
Better add a check like : get_view_query()

			/* A _RETURN rule should have only one action */
			if (list_length(rule->actions) != 1)
				elog(ERROR, "invalid _RETURN rule action specification");

Even we're sure view only has one action.